doc/vlan_stacking/vlan_stacking_HLD.md

```diff
@@ -0,0 +1,635 @@
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID;
+attr.value.u16 = 20;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_PORT;
+attr.value.oid = 30;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_APPLIED_VLAN_ID;
+attr.value.u16 = 10;
+vlan_stacking_entry_attrs.push(attr);
+
+sai_vlan_api->create_vlan_stack(&vlan_stacking_oid, gSwitchId, (uint32_t)vlan_stacking_entry_attrs.size(), vlan_stacking_entry_attrs.data());
+```
+
+# Configuration and management
+
+## CLI/YANG model Enhancements
+
+### CLI
+
+CLI commands for VLAN stacking are not supported yet.
+
+### Yang model
+
+New yang model `sonic-vlan-stacking.yang` is defined to describe VLAN stacking configuration.
+
+```json
+module sonic-vlan-stacking {
+
+    yang-version 1.0;
+
+    namespace "http://github.com/Azure/sonic-vlan-stacking";
+    prefix vlan-stacking;
+
+    import sonic-extension {
+        prefix ext;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+        revision-date 2019-07-01;
+    }
+
+    description "VLAN Stacking YANG Module for SONiC OS";
+
+    revision 2021-12-15 {
+        description "First Revision";
+    }
+
+    container sonic-vlan-stacking {
+        container VLAN_STACKING {
+            description "VLAN_STACKING part of config_db.json";
+
+            list VLAN_STACKING_LIST {
+                key "INTERFACE_NAME STAGE VLAN_ID";
+
+                ext:key-regex-configdb-to-yang "^([a-zA-Z0-9_-]+)|([a-zA-Z0-9_-]+)|([0-9]+)$";
+
+                ext:key-regex-yang-to-configdb "<INTERFACE_NAME>|<STAGE>|<VLAN_ID>";
+
+                leaf INTERFACE_NAME {
+                    type union {
+                        type leafref {
+                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:port_name;
+                        }
+                        type leafref {
+                            path /lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:portchannel_name;
+                        }
+                    }
+                }
+
+                leaf STAGE {
+                    type string {
+                        pattern 'ingress|egress';
+                    }
+                }
+
+                leaf VLAN_ID {
+                    type leafref {
+                        path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid;
+                    }
+                }
+
+                leaf action {
+                    type string {
+                        pattern 'push|pop|swap';
+                    }
+                }
+
+                leaf s_vlanid {
+                    type leafref {
+                        path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid;
```

Comment:
Not sure how this s_vlanid is being used for various actions in different stages, please add description.

Reply:
Provided only if action is push or swap. Will to add comment in here.